### DISCUSSION TITLE
Mapforge Project

### DISCUSSION THREAD

**Discussion by DesenvolvimentoEpf** [mapsforge/mapsforge#1399] [Q&A] [answered]
Hello, I'm new to development and I'm working on a project that displays the offline map and I'm using mapforge to display a map file, I need some help I'm trying to use an offline map file but I get an exception when I run the project, see below my method to import the file and pass it to mapview:

      `  MapView mapView = findViewById(R.id.map);
        mapView.getModel().mapViewPosition.setCenter(new LatLong(52.520008, 13.404954)); 
        mapView.getModel().mapViewPosition.setZoomLevel((byte) 12); 

        File mapFile = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS), "automatta_trator/map/brazil.map");
        

        if (mapFile.exists()){

            MapDataStore mapDataStore = new MapFile(mapFile);
            TileCache tileCache = AndroidUtil.createTileCache(this, new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS), "automatta_trator/map").getAbsolutePath(),
                    mapView.getModel().displayModel.getTileSize(), 1f,
                    mapView.getModel().frameBufferModel.getOverdrawFactor());

            try {

                File renderFile = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS), "automatta_trator/tiles/rendertheme.xsd");

                XmlRenderTheme xmlRenderTheme = new ExternalRenderTheme(renderFile);

                TileRendererLayer tileRendererLayer = AndroidUtil.createTileRendererLayer(tileCache,  mapView.getModel().mapViewPosition, mapDataStore, xmlRenderTheme);
                tileRendererLayer.setTextScale(1.5f);
                mapView.getLayerManager().getLayers().add(tileRendererLayer);

            } catch (FileNotFoundException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }

        }else
        {

            Toast.makeText(context,"Não existe", Toast.LENGTH_SHORT).show();

        }`

i get the following exception:

`W/DatabaseRenderer: Exception: java.lang.IllegalArgumentException: Parse error for XML rendertheme`

could you help me to correct this error to import the map;

**Comment by devemux86** [comment, marked as answer]
It seems the render theme file cannot be read. Android has restrictions on public folders.
See the [GettingStarted](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) example for how to use Mapsforge on Android.

---

If you start now, you can also check [VTM](https://github.com/mapsforge/vtm) (OpenGL) library, compatible with Mapsforge maps.
(see the VTM [GettingStarted](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/GettingStarted.java) example)

**Reply by DesenvolvimentoEpf** [reply]
the xml file I'm using is the mapsforge documentation

**Reply by devemux86** [reply]
Also you are using the theme schema xsd file, not a real theme xml file.

**Reply by DesenvolvimentoEpf** [reply]
I can add polygons and OvelayItem in mapsforge?

**Comment by devemux86** [comment]
> I can add polygons and OvelayItem in mapsforge?

See the [OverlayMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java) example.
